common/src/app/common/geom/shapes.cljc

```diff
@@ -41,11 +41,21 @@
 
 ;; --- Helpers
 
+(defn bounding-box
+  "Returns a rect that wraps the shape after all transformations applied."
+  [shape]
+  ; TODO: perhaps we need to store this calculation in a shape attribute
+  (gpr/points->rect (:points shape)))
+
 (defn left-bound
+  "Returns the lowest x coord of the shape BEFORE applying transformations."
+  ; TODO: perhaps some day we want after transformations, but for the
+  ;       moment it's enough as is now.
   [shape]
   (get shape :x (:x (:selrect shape)))) ; Paths don't have :x attribute
```

Comment:
I think a better approach here would be: 

`(or (:x shape) (:x (:selrect shape)))`

It will avoid executing the second part if the first one success

Reply:
done